orocos_kdl/src/chainjnttojacdotsolver.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+    Computes the Jacobian time derivative
+    Copyright (C) 2015  Antoine Hoarau <hoarau [at] isir.upmc.fr>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+
+#ifndef KDL_CHAINJNTTOJACDOTSOLVER_HPP
+#define KDL_CHAINJNTTOJACDOTSOLVER_HPP
+
+#include "solveri.hpp"
+#include "frames.hpp"
+#include "jntarrayvel.hpp"
+#include "jacobian.hpp"
+#include "chain.hpp"
+#include "framevel.hpp"
+#include "chainjnttojacsolver.hpp"
+#include "chainfksolverpos_recursive.hpp"
+
+namespace KDL
+{
+    
+class ChainJntToJacDotSolver : public SolverI
+{
+/*
+ * Computes the Jacobian time derivative (Jdot) by calculating the partial derivatives
+ * regarding to a joint angle, in the Hybrid, Body-fixed or Inertial representation. 
+ * This work is based on : 
+ * 
+ * Symbolic differentiation of the velocity mapping for a serial kinematic chain
+ * H. Bruyninckx, J. De Schutter
+ * doi:10.1016/0094-114X(95)00069-B
+ * 
+ * url : http://www.sciencedirect.com/science/article/pii/0094114X9500069B
+*/
+public:
+    static const int E_JAC_DOT_FAILED= -100;
+    
+    // Hybrid representation ref Frame: base, ref Point: end-effector
+    static const int HYBRID = 0;
+    // Body-fixed representation ref Frame: end-effector, ref Point: end-effector
+    static const int BODYFIXED = 1;
+    // Intertial representation ref Frame: base, ref Point: base
+    static const int INTERTIAL = 2;
+    
+    explicit ChainJntToJacDotSolver(const Chain& chain);
+    virtual ~ChainJntToJacDotSolver();
+    /**
+     * @brief Computes \f$ {}_{bs}\dot{J}^{ee}.\dot{q} \f$
+     * 
+     * @param q_in Current joint positions and velocities
+     * @param jac_dot_q_dot The twist representing Jdot*qdot
+     * @param seg_nr The final segment to compute
+     * @return int 0 if no errors happened 
+     */
+    virtual int JntToJacDot(const KDL::JntArrayVel& q_in, KDL::Twist& jac_dot_q_dot, int seg_nr = -1);
```

Comment:
@ahoarau I took up the comments in #76. Perhaps you can have a pass at it.

I'm quite sure you need `qdot * qdot` since `d(J(q)) = Jdot(q) * qdot` (function composition).